backend/src/main/java/com/bakdata/conquery/models/datasets/concepts/filters/postalcode/PostalCodesManager.java

```diff
@@ -0,0 +1,102 @@
+package com.bakdata.conquery.models.datasets.concepts.filters.postalcode;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.zip.GZIPInputStream;
+
+import com.github.powerlibraries.io.In;
+import com.univocity.parsers.csv.CsvParser;
+import com.univocity.parsers.csv.CsvParserSettings;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class PostalCodesManager {
+
+	@Getter
+	@Setter
+	private String dataFilePath = null;
+
+	@Getter
+	private List<Double> data;
+
+	private int recordsNumber = 0;
+
+	public PostalCodesManager() {
+		try {
+			loadData();
+		}
+		catch (IOException e) {
+			e.printStackTrace();
+		}
+		catch (URISyntaxException e) {
+			e.printStackTrace();
+		}
+	}
+
+
+	/**
+	 * This method loads the postcodes and their distances between each other
+	 */
+	private void loadData() throws IOException, URISyntaxException {
+		if (getDataFilePath() == null) {
+			setDataFilePath("/com/bakdata/conquery/postalcodes.csv.gz");
+		}
+
+
+		final PostalCodeProcessor rowProcessor = new PostalCodeProcessor();
+
+		final CsvParserSettings csvParserSettings = new CsvParserSettings();
+		csvParserSettings.setDelimiterDetectionEnabled(true);
+		csvParserSettings.setHeaderExtractionEnabled(false);
+		csvParserSettings.setProcessor(rowProcessor);
+
+		final CsvParser parser = new CsvParser(csvParserSettings);
+		parser.parse(new InputStreamReader(new GZIPInputStream(In.resource(getDataFilePath()).asStream()), StandardCharsets.US_ASCII));
+
+
+		data = rowProcessor.getData();
+		recordsNumber = rowProcessor.getRecordsNumber();
+	}
+
+
+	/**
+	 * This method filters out all postcodes that are within the specified distance radius from the specified reference postcode (reference-postcode included).
+	 */
+	public String[] filterAllNeighbours(PostalCodeSearchEntity postalCodeSearchEntity) {
+		try {
+			final double plz = Double.parseDouble(postalCodeSearchEntity.getPlz());
+			if (postalCodeSearchEntity.getRadius() == 0) {
+				return new String[]{String.format("%05d", (int) plz)};
+			}
+
+			final List<String> foundPostalCodesList = new ArrayList<>();
+			for (int i = 0; i < recordsNumber; i++) {
+				final double plz1 = data.get(i);
+				final double plz2 = data.get(++i);
+				final double distance = data.get(++i);
+
```

Comment:
```suggestion
if(plz1 != plz && plz2 != plz) continue;
```
Damit sparst du dir das branching unten.